### DISCUSSION TITLE
Example '04-file transfer' not working on separate machines

### DISCUSSION THREAD

**Discussion by alvarogarciamartinez** [eclipse-edc/Samples#75] [General]
Hello, 
I have tried running the file transfer sample in two different machines, one acting as the provider and one as the consumer, using a third machine to interact with the consumer (the client).  Some minor modifications have been done to the code, to offer more than one asset in the provider.
![image](https://user-images.githubusercontent.com/24429620/169471400-33cb6bdb-b522-4c64-bf3c-ee7bb6b1f62c.png)

After posting a contract and it being accepted, I have posted the transfer process, following the documentation
![image](https://user-images.githubusercontent.com/24429620/169470507-7a5d81cb-b8f4-494c-9860-5e94498036d4.png)

The transfer process is completed, but the file instead of transferring to the dataDestination folder in the consumer it's just copied in the provider machine, in the dataDestionation folder. 

Is this the expected behaviour?

**Comment by ndr-brt** [comment]
as specified in the [first paragraph of the sample-04](https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/blob/db34d77bc86c0013a6f98208fd14f62ff9549dcb/samples/04.0-file-transfer/README.md?plain=1#L3-L7):
> we will run both connectors on the same physical machine (i.e. your development machine) and the file is transferred from one folder in the file system to another folder

so yes, sample-04 is intended to run on a single machine.

**Comment by mspiekermann** [comment]
Using the EDC to exchange files between instances running on different machines is addressed in [sample 5](https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/tree/db34d77bc86c0013a6f98208fd14f62ff9549dcb/samples/05-file-transfer-cloud), so please have a look.
